tsdb/series_partition.go

```diff
@@ -226,6 +224,13 @@ func (p *SeriesPartition) CreateSeriesListIfNotExists(collection *SeriesCollecti
 	newKeyRanges := make([]keyRange, 0, writeRequired)
 	newIDs := make(map[string]SeriesIDTyped, writeRequired)
 
+	// Pre-grow index for large writes.
+	if writeRequired >= 10000 {
```

Comment:
Can you make `10000` the default, but a configurable value. In other setups we may want to tweak this.